File: code/community/1306412665064656903/handling-long-running-whisper-requests.go
// Go code to check for job status using Deepgram API
// Requires installation of Deepgram Go SDK
// To run, replace placeholders with real values and implement your error handling

package main

import (
	"fmt"
	"github.com/deepgram-devs/deepgram-go-sdk"
	"os"
)

func main() {
	apiKey := os.Getenv("DEEPGRAM_API_KEY")
	jobId := "your-job-id"

	client := deepgram.NewClient(apiKey)

	response, err := client.GetTranscriptionStatus(jobId)
	if err != nil {
		fmt.Printf("Error: %v\n", err)
		return
	}

	fmt.Printf("Job status: %v\n", response.Metadata.Status)
	// Handle job status accordingly
}

File: code/community/1306412665064656903/handling-long-running-whisper-requests.cs
// C# code to check for job status using Deepgram API
// Requires installation of `DeepgramSDK` via NuGet
// To run, replace placeholders with real values

using DeepgramSDK;
using System;
using System.Threading.Tasks;

class Program
{
    static async Task Main(string[] args)
    {
        string apiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");
        string jobId = "your-job-id";

        var dgClient = new DeepgramClient(apiKey);

        try
        {
            // Fetch job status
            var response = await dgClient.GetTranscriptionStatusAsync(jobId);
            Console.WriteLine("Job status: {0}", response.Metadata.Status);
            // Handle job status accordingly
        }
        catch (Exception ex)
        {
            Console.WriteLine("Error: {0}", ex.Message);
        }
    }
}